### ISSUE TITLE
`bigip_as3` Read/Exists/Update actions should be restricted to target tenant

### ISSUE THREAD

**Issue by jakauppila** [terraform-providers/terraform-provider-bigip#253]
Currently within the `bigip_as3` resource, the `resourceBigipAs3Read`, `resourceBigipAs3Exists`, and `resourceBigipAs3Update` actions all target `url := client_bigip.Host + "/mgmt/shared/appsvcs/declare"` when they should probably target the desired tenant.
```
name := d.Get("tenant_name").(string)
url := client_bigip.Host + "/mgmt/shared/appsvcs/declare/" + name`
```
This should trim down the response body to only the tenant that is being modified.

While it's not an issue right now, with how we're planning to deploy our application with AS3, we're going to end up having thousands of tenants on the appliances.

**Comment by focrensh** [comment]
Tracking with TER-279

**Comment by focrensh** [comment]
Ideally the update/exists/read should be limited to the Tenants that were within the declaration itself. They should not need to be specified by the user as they are within the declaration.

**Comment by jakauppila** [comment]
I suppose I hadn't considered that someone may hold multiple tenants within a single declaration; seems counter intuitive to what AS3 is aiming to achieve.

That being said, what value does the `tenant_name` property hold if not being used to scope the REST API calls?

**Comment by focrensh** [comment]
AS3 is declarative for the entire config, but with the updateMode to selective(default) it only updates the current Tenants specified in the current declaration.

You are correct it is being used for Scope currently. The declaration itself could be auto-populated based on the original Declaration. 

- Apply: T1, T2
- Destroy: (Since T1 and T2 were in original declaration, it knows to only Delete those without the user having to explicitly specify)

**Comment by jakauppila** [comment]
Right; so it's some of those core design decision that raises some concerns for me about AS3 in the Enterprise since it's just a couple characters away from someone to change that updateMode to `complete` or to execute a DELETE against the `/mgmt/shared/appsvcs/declare` while accidentally omitting a tenant.

To clarify/scope things appropriately, would there be value in creating a new `bigip_as3_tenant` resource that's more of an opinionated/pointed method of deployment that ensures that a single tenant is being targeted (likely involving building the declaration with the supplied tenant definition).

**Comment by RavinderReddyF5** [comment]
Fixed in v1.2.0
